api/placer.py

```diff
@@ -93,8 +93,9 @@ def save_file(self, field=None, file_attrs=None):
         if file_attrs is not None:
             self.container = hierarchy.upsert_fileinfo(self.container_type, self.id_, file_attrs)
 
-            # Queue any jobs as a result of this upload
-            rules.create_jobs(config.db, self.container, self.container_type, file_attrs)
+            # Queue any jobs as a result of this upload, uploading to a gear will not make jobs though
+            if self.container_type != 'gear':
```

Comment:
Instead of changing this logic, I just wouldn't pass the `file_attrs` to `save_file`. You also don't want to `upsert_fileinfo`, that attaches file objects to containers.